File: src/main/java/com/gabrielczar/springvalidations/repositories/UserRepository.java
package com.gabrielczar.springvalidations.repositories;

import com.gabrielczar.springvalidations.domain.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.repository.query.Param;
import org.springframework.data.rest.core.annotation.RepositoryRestResource;

import java.util.List;

@RepositoryRestResource(collectionResourceRel = "users", path = "users")
public interface UserRepository extends JpaRepository<User, Integer> {
    List<User> findByName(@Param("name") String name);
}

File: src/main/kotlin/com/gabrielczar/simpleappvalidation/repositories/UserRepository.kt
package com.gabrielczar.simpleappvalidation.repositories

import com.gabrielczar.simpleappvalidation.domain.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import javax.transaction.Transactional

@Repository
@Transactional
interface UserRepository : JpaRepository<User, Long>
